frontend/src/views/producer/dashboard/edition-form/CoverImage.jsx

```diff
@@ -20,15 +21,47 @@ const FlexBox = styled.div`
   width: 100%;
   align-items: center;
 `;
-
+const ErrorMessage = styled('span')`
+  p {
+    display: flex;
+    font-size: 1rem;
+    margin: 0;
+    text-align: left;
+    color: red;
+    display: ${props => (props.show ? 'flex' : 'none')};
+  }
+`;
 class CoverImage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.uploadImage = this.uploadImage.bind(this);
+  }
+
+  async uploadImage(ev) {
+    ev.preventDefault();
+    this.props.uploadProgress(true);
+    const imageurl = await this.props.store.uploadImage(ev.target.files[0]);
```

Comment:
I don't object to this way of doing but i designed the store to be a global, commited state. draft states like temporarely uploaded images should be kept local to the component.